### PR TITLE
[Merged by Bors] - fix: Add prompt for project group in `smdk generate`

### PR DESCRIPTION
**Pull request by tjtelan**
Added prompt to ask for a group name if we don't have one set.

Requires login to InfinyOn Cloud (Tested with with dev instances though)

When asked the first time, we set the hub id with the Hub service. Then future packages will use that value as default.

```
./target/debug/smdk generate \
--with-params \
--sm-crate-path $(pwd)/crates/fluvio-smartmodule \
--template-path $(pwd) \
--destination $(mktemp -d) \
--sm-type map \
--hub-remote https://hub-dev.infinyon.cloud
my-new-sm-package
```

It was hard to test, but I'm certain that if a group name is taken, the Hub will return an error when trying to set the ID

```
Generating new SmartModule project: hellogroup
smartmodule-type => 'map'
smartmodule-params => 'true'
fluvio-smartmodule-cargo-dependency => '{ path = "/Users/telant/Documents/fluvio/crates/fluvio-smartmodule" }'
🔧   Destination: /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.Y7AiGoHa/hellogroup ...
🔧   Generating template ...
🤷   Please set a group name : tj
Ignoring: /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/.tmpEyPoIq/smartmodule/cargo_template/cargo-generate.toml
[1/5]   Done: Cargo.toml
[2/5]   Done: README.md
[3/5]   Done: SmartModule.toml
[4/5]   Done: src/lib.rs
[5/5]   Done: src
🔧   Moving generated files into: `/var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.Y7AiGoHa/hellogroup`...
💡   Initializing a fresh Git repository
✨   Done! New project created /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.Y7AiGoHa/hellogroup
hub: hubid tj is set
hubid set to tj
```

### CODE REVIEW

File: crates/smartmodule-development-kit/src/generate.rs [hunk 1]
```diff
@@ -166,6 +176,10 @@ pub struct GenerateOpt {
         "SmCrateSourceGit", "SmCrateSourceCratesIo", "SmCrateSourcePath"],)]
     develop: bool,
 
+    /// Set the remote URL for the hub
+    #[clap(long, hide_short_help = true)]
+    hub_remote: Option<String>,
```
Comment (sehz) [inline crates/smartmodule-development-kit/src/generate.rs:181]: remove this.  use env var

**Review by sehz — CHANGES_REQUESTED** [review]
First pass review

**Review by sehz — CHANGES_REQUESTED** [review]
group need to be specified

**Review by sehz — CHANGES_REQUESTED** [review]
init code is not generated

File: crates/smartmodule-development-kit/src/generate.rs (hunk 1 above)
Comment (tjtelan) [reply crates/smartmodule-development-kit/src/generate.rs:181]: It is hidden from users, and only used in testing. An env var didn't behave well in the CI testing

**Comment by tjtelan** [comment]
> init code is not generated

Can you be a lot more specific about what you're seeing? Because if you're referring to `smdk generate` not generating code, then I disagree.

```shell
$ ./target/debug/smdk generate --destination $(mktemp -d)
project-group => 'my-group'
fluvio-smartmodule-cargo-dependency => '"0.2.5"'
🤷   Project Name : projectname
🔧   Destination: /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.m3fYl445/projectname ...
🔧   Generating template ...
✔ 🤷   Will your SmartModule use init parameters? · true
✔ 🤷   Which type of SmartModule would you like? · map
Ignoring: /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/.tmpxMEyMg/cargo-generate.toml
[1/5]   Done: Cargo.toml
[2/5]   Done: README.md
[3/5]   Done: SmartModule.toml
[4/5]   Done: src/lib.rs
[5/5]   Done: src
🔧   Moving generated files into: `/var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.m3fYl445/projectname`...
💡   Initializing a fresh Git repository
✨   Done! New project created /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.m3fYl445/projectname
```

``` shell
$ ls -alh /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.m3fYl445/projectname
total 24
drwxr-xr-x  7 telant  staff   224B Oct 24 10:22 .
drwx------  3 telant  staff    96B Oct 24 10:22 ..
drwxr-xr-x  9 telant  staff   288B Oct 24 10:22 .git
-rw-r--r--  1 telant  staff   309B Oct 24 10:22 Cargo.toml
-rw-r--r--  1 telant  staff   1.8K Oct 24 10:22 README.md
-rw-r--r--  1 telant  staff   190B Oct 24 10:22 SmartModule.toml
drwxr-xr-x  3 telant  staff    96B Oct 24 10:22 src
```

```
$ cat /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.m3fYl445/projectname/SmartModule.toml 
[package]
name = "projectname"
group = "my-group"
version = "0.1.0"
apiVersion = "0.1.0"
description = ""
license = "Apache-2.0"

[[params]]
name = "input"
description = "input description"
```

File: crates/smartmodule-development-kit/src/generate.rs (hunk 1 above)
Comment (sehz) [reply crates/smartmodule-development-kit/src/generate.rs]: ok. fair enough

**Comment by sehz** [comment]
> > init code is not generated
> 
> Can you be a lot more specific about what you're seeing? Because if you're referring to `smdk generate` not generating code, then I disagree.
> 
> ```shell
> $ ./target/debug/smdk generate --destination $(mktemp -d)
> project-group => 'my-group'
> fluvio-smartmodule-cargo-dependency => '"0.2.5"'
> 🤷   Project Name : projectname
> 🔧   Destination: /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.m3fYl445/projectname ...
> 🔧   Generating template ...
> ✔ 🤷   Will your SmartModule use init parameters? · true
> ✔ 🤷   Which type of SmartModule would you like? · map
> Ignoring: /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/.tmpxMEyMg/cargo-generate.toml
> [1/5]   Done: Cargo.toml
> [2/5]   Done: README.md
> [3/5]   Done: SmartModule.toml
> [4/5]   Done: src/lib.rs
> [5/5]   Done: src
> 🔧   Moving generated files into: `/var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.m3fYl445/projectname`...
> 💡   Initializing a fresh Git repository
> ✨   Done! New project created /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.m3fYl445/projectname
> ```
> 
>  ```shell
> $ ls -alh /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.m3fYl445/projectname
> total 24
> drwxr-xr-x  7 telant  staff   224B Oct 24 10:22 .
> drwx------  3 telant  staff    96B Oct 24 10:22 ..
> drwxr-xr-x  9 telant  staff   288B Oct 24 10:22 .git
> -rw-r--r--  1 telant  staff   309B Oct 24 10:22 Cargo.toml
> -rw-r--r--  1 telant  staff   1.8K Oct 24 10:22 README.md
> -rw-r--r--  1 telant  staff   190B Oct 24 10:22 SmartModule.toml
> drwxr-xr-x  3 telant  staff    96B Oct 24 10:22 src
> ```
> 
>  ```
> $ cat /var/folders/r8/4x6_d2rn283946frzd1gc1pr0000gn/T/tmp.m3fYl445/projectname/SmartModule.toml 
> [package]
> name = "projectname"
> group = "my-group"
> version = "0.1.0"
> apiVersion = "0.1.0"
> description = ""
> license = "Apache-2.0"
> 
> [[params]]
> name = "input"
> description = "input description"
> ```

If init is specified then, init skeleton must be generated otherwise, it is misleading and not complete:
```
#[smartmodule(init)]
fn init(params: SmartModuleExtraParams) -> Result<()> {
    if let Some(regex) = params.get("input") {
        todo!("complete this")
    } else {
        Err(SmartModuleInitError::MissingParam("input".to_string()).into())
    }
}
```

```

**Review by sehz — CHANGES_REQUESTED** [review]
Still need to fill int the init code

**Comment by digikata** [comment]
Tested from blank hub id credentials and w/ creds. Works for me. (init code aside)

**Review by sehz — APPROVED** [review]
group prompt works for me.  had previous hub id set.
we can defer init generation

**Pull request closed without merging**